archive/d/dg/file-io.dg

```diff
@@ -0,0 +1,22 @@
+import '/os'
+
+write_file = filename s ->
+    except
+        err =>
+            open filename 'w' |>.write <| s
+        err :: OSError =>
+            print "Cannot open file" filename
+
+read_file = filename ->
+    except
+        err =>
+            with fd = open filename 'r' =>
+                print $ fd.read!
+        err :: OSError =>
+            print "Cannot open file" filename
+
+
+filename = "tmp_file"
```

Comment:
For consistency, I think we call the file `output.txt`. See the [Contributing Doc](https://github.com/TheRenegadeCoder/sample-programs/blob/master/CONTRIBUTING.md#file-io-rules).